control-plane-operator/controllers/hostedcontrolplane/pki/kas.go

```diff
@@ -155,3 +161,17 @@ func generateKubeConfig(url string, crtBytes, keyBytes, caBytes []byte) ([]byte,
 func inClusterKASURL(namespace string, apiServerPort int32) string {
 	return fmt.Sprintf("https://%s:%d", manifests.KubeAPIServerServiceName, apiServerPort)
 }
+
+func addBracketsIfIPv6(apiAddress string) (string, error) {
+
+	if utilsnet.IsIPv6String(apiAddress) {
+		return fmt.Sprintf("[%s]", apiAddress), nil
+	}
+
+	// if contains a "." it could mean that we have an URL or IPv4
+	if strings.Contains(apiAddress, ".") {
+		return apiAddress, nil
+	}
+
+	return "", fmt.Errorf("unknown url")
```

Comment:
We should just `return apiAddress` on line 170.  This creates a new error path where there was not one previously. A url like `http://host` would hit the error path but is valid.

Reply:
Thanks, makes sense.